### PR TITLE
LPD-34380 - Organization api

### PR DESCRIPTION
**Pull request by matyaswollner**
https://liferay.atlassian.net/browse/LPD-34380

### CODE REVIEW

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
Cc @smottal

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/OrganizationResourceTest.java [hunk 1]
```diff
@@ -403,7 +494,7 @@ public void testPutOrganizationByExternalReferenceCode() throws Exception {
 
 	@Override
 	protected String[] getAdditionalAssertFieldNames() {
-		return new String[] {"name"};
+		return new String[] {"name", "externalReferenceCode"};
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/OrganizationResourceTest.java:497]: sort these

**Comment by smottal** [comment]
@matyaswollner a lot of changes but are pretty much the same multiple times

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:report:65639208

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java [hunk 2]
```diff
@@ -335,6 +336,22 @@ public UserAccount getMyUserAccount() throws Exception {
...
+	@Override
+	public Page<UserAccount>
+			getOrganizationByExternalReferenceCodeUserAccountsPage(
+				String externalReferenceCode, String search, Filter filter,
+				Pagination pagination, Sort[] sorts)
+		throws Exception {
+
+		Organization organization =
+			_organizationService.getOrganizationByExternalReferenceCode(
+				contextCompany.getCompanyId(), externalReferenceCode);
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java:348]: argh... probably we need to work on this 😢

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:report:66863737

**Pull request closed without merging**